src/main/java/me/zeroeightsix/kami/util/VectorUtil.java

```diff
@@ -0,0 +1,33 @@
+package me.zeroeightsix.kami.util;
+
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+
+import java.util.List;
+
+public class VectorUtil {
+
+    public static List<Vec3d> getVecsInArea(Vec3d pos1, Vec3d pos2) {
+        List<Vec3d> returnList = null;
+
+        int minX = (int) Math.round(Math.min(pos1.x, pos2.x));
+        int maxX = (int) Math.round(Math.max(pos1.x, pos2.x));
+
+        int minY = (int) Math.round(Math.min(pos1.y, pos2.y));
+        int maxY = (int) Math.round(Math.max(pos1.y, pos2.y));
+
+        int minZ = (int) Math.round(Math.min(pos1.z, pos2.z));
+        int maxZ = (int) Math.round(Math.max(pos1.z, pos2.z));
+
+        for (int x = minX; x < maxX; x++) {
+            for (int y = minY; y < maxY; y++) {
+                for (int z = minZ; z < maxZ; z++) { returnList.add(new Vec3d(x, y, z)); }
```

Comment:
I think you might have meant to do this with blockpos rather than vec3d. Vectors are used when floating point precision is needed, blockposes are used when precision is only necessary to the nearest block/meter

Reply:
in this instance its the same thing

Reply:
ill make an overload

Reply:
No it isnt the same thing. Using floats, as vectors do, uses an unnecessary amount of memory in this context

Reply:
vectors use doubles, not floats

Reply:
anyways i removed it from that function and replaced it with blockpos return but still has an overload for calling with vectors

Reply:
Ah sorry thought you were using Vec2f. That is even worse, doubles are 64 bit compared to 32 bit floats.

Reply:
yeah i know its called double for a reasonn lol